tests/hooks/test_model_hooks.py

```diff
@@ -0,0 +1,550 @@
+                        '{SOURCE}.{{{{ target.schema }}}}' as target_schema,
+                        '{{{{ target.type }}}}' as target_type,
+                        '{{{{ target.user }}}}' as target_user,
+                        '{{{{ target.get(pass, "") }}}}' as target_pass,
+                        {{{{ target.threads }}}} as target_threads,
+                        '{{{{ run_started_at }}}}' as run_started_at,
+                        '{{{{ invocation_id }}}}' as invocation_id
+                        from {{{{ ref('post') }}}}""".strip()
+                        ],
+                    }
+                },
+            }
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "hooked.sql": models__hooked,
+            "post.sql": models__post,
+            "pre.sql": models__pre,
+        }
+
+    def test_pre_post_model_hooks_refed(self, project, dbt_profile_target):
+        run_dbt()
+        self.check_hooks("start", project, dbt_profile_target.get("host", None))
+        self.check_hooks("end", project, dbt_profile_target.get("host", None))
+
+
+class TestPrePostModelHooksOnSeeds(object):
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"example_seed.csv": seeds__example_seed_csv}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": properties__seed_models}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "seed-paths": ["seeds"],
+            "models": {},
+            "seeds": {
+                "post-hook": [
+                    "alter table {{ this }} add columns (new_col int)",
+                    "update {{ this }} set new_col = 1",
+                    # call any macro to track dependency: https://github.com/dbt-labs/dbt-core/issues/6806
+                    "select CAST(1 AS {{ dbt.type_int() }}) as id",
+                ],
+                "quote_columns": False,
+            },
+        }
+
+    def test_hooks_on_seeds(self, project):
+        res = run_dbt(["seed"])
+        assert len(res) == 1, "Expected exactly one item"
+        res = run_dbt(["test"])
+        assert len(res) == 1, "Expected exactly one item"
+
+
+class TestHooksRefsOnSeeds:
+    """
+    This should not succeed, and raise an explicit error
+    https://github.com/dbt-labs/dbt-core/issues/6806
+    """
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"example_seed.csv": seeds__example_seed_csv}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": properties__seed_models, "post.sql": models__post}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "seeds": {
+                "post-hook": [
+                    "select * from {{ ref('post') }}",
+                ],
+            },
+        }
+
+    def test_hook_with_ref_on_seeds(self, project):
```

Comment:
I'm unsure what you are referring to here. This is a pytest test function, were you referring to a different fixture override?

Reply:
I guess a better question would be, why are we not inheriting the tests instead of copying everything over? And if we need to make changes to accommodate dremio, why don't we just copy over the necessary fixtures, rather than all the fixtures?

Reply:
A lot of these fixtures have been changed to accommodate dremio - some might not even be directly obvious as they use fixtures that have been imported from a local dbt-dremio fixtures file with the same name. If you look at the imports, unchanged fixtures are being imported from dbt core whenever possible.

Reply:
If you're inheriting the test classes, you wouldn't need to import the unchanged fixtures over.

Reply:
I have changed it to inherit the test from dbt-core while only updating the fixture.